### ISSUE TITLE
Build Glossary of terms (in english and arabic)

### ISSUE THREAD

**Issue by des-des** [foundersandcoders/master-reference#194]
So this will be useful for students who speak English as a first language as well as Arabic speakers who do not. Can we start by making a list of words, then we can write definitions in a pull request.

Post words that should be in the glossary below.

**Comment by des-des** [comment]
Okay am sure I have missed some people but this should get us started

**Comment by mattlub** [comment]
render

**Comment by mattlub** [comment]
merge

**Comment by des-des** [comment]
architecture, server, client, browser, back-end, front-end, api, rest-api, tdd

**Comment by Jbarget** [comment]
request, response, validation, security (in terms of tech), framework, library

**Comment by jsms90** [comment]
@Cleop keywords - your fave :wink:

**Comment by Cleop** [comment]
Ajax, semantic, transpiler, user story, responsive (design), prototype, method, framework, library, closure, recursion, function expression, currying, comet, box model, agile, factories, Java VM, BEAM/ Erlang VM.

The list goes on and on... and potentially some of these things can't be just explained in a glossary. I'm just throwing these out as a starting point.

**Comment by Jbarget** [comment]
I'd say we dont need Java VM, BEAM or Erlang VM they seem a bit advanced n I think they're too relevant to the 16 week course

whats comet? (had a quick google n is it related to Meteor)?

**Comment by Cleop** [comment]
Agreed, @Jbarget. I was just throwing some out there from my list of key terms and wasn't sure how broad the glossary was meant to be.

Comet is a word for streaming : https://en.wikipedia.org/wiki/Comet_(programming)

Oh another 2: dynamic / static languages

**Comment by mattlub** [comment]
@des-des could you clear up exactly what this will be/how it will be used when it's finished?
An online list which will be provided as a resource?

**Comment by des-des** [comment]
@mattlub I would like to have this in place for the next cohort, but will need a translator. Would be very happy if someone was interested in making a start on this.

**Comment by HilbertSpitzer** [comment]
https://www.almaany.com/en/dict/ar-en/facilitator/
It's a pretty good site, it can properly define things that I haven't seen other sites being able to properly define, it also has Computer and Technology terms.
For example, here's the definition of execute when you use exclusive Computer terms
https://www.almaany.com/en/dict/ar-en/execute/?c=Computer

> @mattlub I would like to have this in place for the next cohort, but will need a translator. Would be very happy if someone was interested in making a start on this.

Well, the next cohort is already doing a prep-course. So I think this needs to be re-vitalized
So~o, what does it mean to start this?

**Comment by mattlub** [comment]
Made a [google doc](https://docs.google.com/spreadsheets/d/1engtWbqoHi980n6sLB02TRqRBu3grg5GKCkPEq1Zbvw/edit?usp=sharing) with (nearly) all the words on here ready to be translated!

@HilbertSpitzer 
(also @Suuuuha @shiryz @Karyum)

**Comment by RanaAlqrenawi** [comment]
This is great and very helpful too. Having list of keyword itself something very good and can support the students while some of the words might be better to stay in English. What makes more sense is to clarify them in Arabic - not necessary translate the word itself - so, I think the Arabic Description column is very valuable. 

I can help in the translation to Arabic. I also can say that having an app for that is something will be good. For example, weekly project for adding new keywords with the En and Ar Description to enrich the list, and having someone to check the technical description and Arabic translation as well before the final submission.

**Comment by shiryz** [comment]
@sajedazoabi, @KatiaAsh, do you have anything to add to @mattlub 's list or translate some?
@veredrec would you like to work on the hebrew translation?

**Comment by veredrec** [comment]
@shiryz sure, I would love to! I added the Hebrew titles to the doc and I will work on it.

**Comment by KatiaAsh** [comment]
@shiryz  I can also work on Arabic section :blush:
